Fix/eslint

HosokawaR opened the pull request:
## 背景

Resolves #24 
その他の Warn / Error が残っている

## 解決策

細かいので各コミットメッセージを参照

## Merge について

https://github.com/twin-te/manaba-report-integration/pull/51 が Merge されてから本 PR を Merge する

HosokawaR committed:
Fix eslintignore

HosokawaR committed:
Add the no-console rule

For error inspection in production, allow `console.error`

HosokawaR committed:
Remove extra `console.log`

HosokawaR committed:
Remove unused variables

HosokawaR commented:
@azarashi2931 

このブランチの `yarn lint` を手元で試すと以下のように Failed するのだけど、CI の `yarn lint` は Passed するのはなんでなんだろう？

```
$ yarn lint
yarn run v1.22.19
$ eslint . --ext .tsx,.ts --fix
Warning: React version not specified in eslint-plugin-react settings. See https://github.com/yannickcr/eslint-plugin-react#configuration .

/home/.../twinte/manaba-report-integration/src/ui/blacklistUtils.ts
  0:0  error  Parsing error: Cannot read properties of undefined (reading 'map')

✖ 1 problem (1 error, 0 warnings)

error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
$ echo $?
1
```

azrsh commented:
Reviewdog のフィルタの影響で、その PR で変更された行以外にエラーがあってもエラーが出ない
https://github.com/reviewdog/reviewdog#added-default

master ブランチの CI ではこれを無効化した方がいいかもね

HosokawaR commented:
なるほどー
最初から lint error があるコードが悪いので、仕方がない

> master ブランチの CI ではこれを無効化した方がいいかもね

たしかに、賛成です。

azrsh commented:
これは別の PR で直すんでとりあえず今あるの lint error は直しちゃってほしい

HosokawaR committed:
Update @typesccript-eslint packages to 4.7

Since @typescript-eslint supports TypeScript 4.1 since 4.7
@typescript-eslint 2.31 occurs parser error.

See https://github.com/typescript-eslint/typescript-eslint/releases/tag/v4.7.0

HosokawaR committed:
Delete "@typescript-eslint/camelcase" rules

This rule has been deprecated.
See https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/camelcase.md

HosokawaR committed:
"explicit-module-boundary-types" rule may be unnesesarry.

Acutually, this rule default has been turnd off and not recommend in typescript eslint 5.0.0

Fly https://github.com/typescript-eslint/typescript-eslint/issues/3746

HosokawaR committed:
Remove extra bangs

HosokawaR commented:
["explicit-module-boundary-types" rule may be unnesesarry.](https://github.com/twin-te/manaba-report-integration/pull/42/commits/f0c24eac61026734f688c6ee44abf53ee6d221c3) って、コミットメッセージの一行目が感想なのおもろいな（）

HosokawaR commented:
> これは別の PR で直すんでとりあえず今あるの lint error は直しちゃってほしい

待たせてすみません
パッケージのバージョン周りで詰まっていたので遅れました

azrsh commented on .eslintrc.json:
```diff
@@ -15,6 +15,7 @@
   },
   "rules": {
     "@typescript-eslint/no-explicit-any": "off",
+    "@typescript-eslint/explicit-module-boundary-types": "off",
```
これオフにしてるのは何で？

HosokawaR replied:
https://github.com/twin-te/manaba-report-integration/pull/42/commits/f0c24eac61026734f688c6ee44abf53ee6d221c3 のコミットメッセージに書いてある、が改めて書くと

まず現状では `@typescript-eslint/explicit-module-boundary-types` のエラーが多くでる。
とはいえこの警告は必要以上に厳しすぎると考えたので `OFF` にした。

自分の意見として外部から参照されるモジュールではなので、明示的に書かなくても TSC の型検査で十分だと考えた。
またそれを補強する事実として、これらのルールは OFF でもよいという意見がそれなりにあり、実際に v5.0.0 では Default で OFF になっている。

Fly https://github.com/typescript-eslint/typescript-eslint/issues/3746

azrsh replied:
OK
単純にエラーが出るので消しましたとかではなくてオフにする根拠があるならいいと思う

azrsh commented on src/background/background.ts:
```diff
@@ -11,7 +11,7 @@ function broadcastMessage(msg: Message): void {
   chrome.runtime.sendMessage(msg)
   chrome.tabs.query({ active: true, currentWindow: true }, (tabs) =>
     // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-    chrome.tabs.sendMessage(tabs[0].id!!, msg)
```
将来的には non null assertion も排除したい

HosokawaR replied:
わかるよ

HosokawaR committed:
Merge branch 'master' into fix/eslint

; Conflicts:
;	src/popup/settingSection.tsx
;	yarn.lock
